### PR TITLE
Chore: move env test is flaky - use never deployed env

### PR DESCRIPTION
**Pull request by chpl**
### Issue & Steps to Reproduce / Feature Request
the test is flaky - often fails with 
```
│ Error: failed to move environment to project id 'e46ce3e9-79b5-4e4d-bba9-1d9b9f252155': 400 Bad Request: cannot move environment while in progress. id: 4cc847e7-c45d-4389-b49d-4e9bdf94f80a
```
(was added in https://github.com/env0/terraform-provider-env0/pull/939)

### Solution
use never deployed environment

### CODE REVIEW

File: tests/integration/012_environment/main.tf [hunk 1]
```diff
@@ -55,7 +55,7 @@ resource "env0_environment" "example" {
   depends_on    = [env0_template_project_assignment.assignment]
```
Comment (yarivg) [inline tests/integration/012_environment/main.tf:55]: Is this environment still necessary?

**Review by yarivg — APPROVED** [review]
seems good

File: tests/integration/012_environment/main.tf (hunk 1 above)
Comment (chpl) [reply tests/integration/012_environment/main.tf:55]: yes. tomer just added the move part [here](https://github.com/env0/terraform-provider-env0/pull/939)

Comment (tomer-landesman) [reply tests/integration/012_environment/main.tf:55]: I think the original purpose of the test was to test different types / settings of environments, and here there was an attempt to kill 2 birds with one stone by testing both regular deployment and move environment. so I think it is still needed for the original purpose

**Pull request merged**